dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -171,7 +170,7 @@ public Route getRouteWithDecryptedAuth(@Nonnull String id) {
    */
   public ResponseEntity<StreamingResponseBody> execute(
       Route route, UserDetails userDetails, Optional<String> subPath, HttpServletRequest request)
-      throws IOException, BadRequestException {
+      throws Exception {
```

Comment:
A service throwing a non-specific exception feels a bit off. Can't we narrow it down to a set of specific ones?
I agree that we don't want a service to throw an `IOException` either but something more semantic that translates to a HTTP response code. But I assume this might be a case where this would require creating more exceptions which I also don't think we should do. Just want to double-check to have you consider this once more.